test/setTokenFactory.spec.ts

```diff
@@ -0,0 +1,105 @@
+import * as chai from "chai";
+import * as _ from "lodash";
+
+import * as ABIDecoder from "abi-decoder";
+import { BigNumber } from "bignumber.js";
+
+// Types
+import { Address } from "../types/common.js";
+
+// Contract types
+import { StandardTokenContract } from "../types/generated/standard_token";
+import { StandardTokenMockContract } from "../types/generated/standard_token_mock";
+import { SetTokenFactoryContract } from "../types/generated/set_token_factory";
+
+// Artifacts
+const SetTokenFactory = artifacts.require("SetTokenFactory");
+
+// Core wrapper
+import { CoreWrapper } from "./utils/coreWrapper";
+
+// Testing Set up
+import { BigNumberSetup } from "./config/bignumber_setup";
+import ChaiSetup from "./config/chai_setup";
+BigNumberSetup.configure();
+ChaiSetup.configure();
+const { expect, assert } = chai;
+
+import { assertTokenBalance, expectRevertError } from "./utils/tokenAssertions";
+import { STANDARD_INITIAL_TOKENS } from "./constants/constants";
+
+contract("SetTokenFactory", (accounts) => {
+  const [
+    authorizedAccount,
+    nonAuthorizedAccount,
+  ] = accounts;
+
+  let mockToken: StandardTokenMockContract;
+  let mockTokens: StandardTokenMockContract[] = [];
+  let setTokenFactory: SetTokenFactoryContract;
+
+  const coreWrapper = new CoreWrapper(authorizedAccount, authorizedAccount);
+
+  before(async () => {
+    ABIDecoder.addABI(SetTokenFactory.abi);
+  });
+
+  after(async () => {
+    ABIDecoder.removeABI(SetTokenFactory.abi);
+  });
+
+  describe("#create", async () => {
+    let caller: Address = authorizedAccount;
+    let components: Address[] = [];
+    let units: BigNumber[] = [];
+    let naturalUnit: BigNumber = new BigNumber(0);
+    let name: string = "";
```

Comment:
It doesn't seem like you're manipulating these in nested test cases, just add it directly to the subject? Just give a default name/symbol instead of empty string.